### DISCUSSION TITLE
I see that the Master.cs input switch has a case time sync, but that results in a [NO_FUNC_CODE_SUPPORT] from the outstation.

### DISCUSSION THREAD

**Discussion by mattheweby** [stepfunc/dnp3#344] [Q&A] [answered]
I'm not new to DNP3, just to your library. I will say that my last deep dive into the protocol was many years ago, so the various group numbers and specific byte patterns are out of my head at the moment. Apologies ahead of time for stupid or nonsensical questions.

I'm currently prototyping an outstation that will essentially maintain a database table for outside access of the binary value collection's states.

I see that the "lts" input to the master CLI causes a RecordCurrentTime, Write, and then g50v3 in the outstation but then I see the [NO_FUNC_CODE_SUPPORT] response and an error in the master CLI. Is this a bug in the example or is there code that I need to implement to complete the exchange successfully?

**Comment by jadamcrain** [comment]
Hi @mattheweby. Which example (language) are you using? I can then point you in the write direction. I believe that, by default, the outstation examples all reject time write requests.

**Reply by mattheweby** [reply]
C#, I haven't taken the time to ingest Rust yet. 😅

Since it rejects them by default I assume I'll need to write an implementation of g50v3 (i didn't think the object 50 had a variation 3, only 1?) into the control handler class?

**Comment by jadamcrain** [comment, marked as answer]
Change this method to return `WriteTimeResult.Ok`:

https://github.com/stepfunc/dnp3/blob/7f8bcf1ee9c2ca6b7a0ce2b682651841e57331e1/ffi/bindings/dotnet/examples/outstation/Program.cs#L23

There's a section in the docs here that discusses what is happening:

https://docs.stepfunc.io/dnp3/1.5.2/guide/docs/api/outstation/application#time-synchronization

Our library will **NOT** change the system clock. It's up to you to do so or reconcile a clock with the time received via DNP3 and then apply a calculated time when updating the database.

**Reply by mattheweby** [reply]
Thank you for your time. I'll do some more reading. We'll also be reaching out to you guys about licensing prior to implementation.

**Comment by jadamcrain** [comment]
No problem at all. Reach out as you have questions.
